crates/sui-types/src/crypto.rs

```diff
@@ -1331,4 +1336,36 @@ impl SignatureScheme {
             SignatureScheme::Secp256k1 => 0x01,
         }
     }
+
+    pub fn from_flag(flag: &str) -> Result<SignatureScheme, SuiError> {
+        match flag {
+            "0" => Ok(SignatureScheme::ED25519),
```

Comment:
I'd suggest that you match against our flags https://github.com/MystenLabs/sui/blob/d352fc2a25320f8082a2960525f621aea4a6b5e1/crates/sui-types/src/crypto.rs#L1333 vs. hardcoding "0" and "1" here, so if we ever change the flags there, the keytool will automatically work with the new assignment. 
*You will just need byte to string conversion which is easy.

Reply:
good point (i went back and forth on this bc it involves validating the string to strictly one byte), revised!